modules/ingest-user-agent/src/main/java/org/elasticsearch/ingest/useragent/UserAgentParser.java

```diff
@@ -145,19 +160,17 @@ private Pattern compilePattern(String regex, String regex_flag) {
         return devicePatterns;
     }
 
-    String getName() {
-        return name;
-    }
-
     public Details parse(String agentString) {
         Details details = cache.get(name, agentString);
 
         if (details == null) {
             VersionedName userAgent = findMatch(uaPatterns, agentString);
             VersionedName operatingSystem = findMatch(osPatterns, agentString);
             VersionedName device = findMatch(devicePatterns, agentString);
+            String deviceType = deviceTypeParser.findDeviceType(agentString, userAgent, operatingSystem, device);
 
-            details = new Details(userAgent, operatingSystem, device);
+
+            details = new Details(userAgent, operatingSystem, device, deviceType);
```

Comment:
Unnecessary newline:
```suggestion
            details = new Details(userAgent, operatingSystem, device, deviceType);
```